rotkehlchen/api/rest.py

```diff
@@ -2697,117 +2697,116 @@ def refresh_evm_transactions(
+            success = False
+        except InputError as e:
+            status_code = HTTPStatus.CONFLICT
+            message = f'Failed to request evm transaction decoding due to {e!s}'
+            success = False
 
-        return {'result': result, 'message': message, 'status_code': status_code}
+        return {'result': success, 'message': message, 'status_code': status_code}
 
     @async_api_call()
     def decode_evmlike_transactions(
             self,
-            data: list[EvmTransactionDecodingApiData],
+            tx_hash: EVMTxHash,
+            chain: EvmlikeChain,  # pylint: disable=unused-argument
     ) -> dict[str, Any]:
         """
-        Decode a set of transactions selected by their transaction hash. If the tx_hashes
-        value is None all the transactions for that chain in the database will be
-        attempted to be decoded. If the tx_hashes argument is provided then the USD
-        price for their events will be queried.
+        Repull all data and redecode events for a single emvlike transaction hash.
+        Chain can only be zksync lite for now.
         """
         task_manager = self.rotkehlchen.task_manager
-        result = None
-        message = ''
-        status_code = HTTPStatus.OK
+        assert task_manager, 'task manager should have been initialized at this point'
+        success, message, status_code = True, '', HTTPStatus.OK
         tracked_addresses = self.rotkehlchen.chains_aggregator.accounts.zksync_lite
-        for entry in data:  # we cheat. Can only be zksync lite for now
-            if entry['tx_hashes'] is None:
-                transactions = self.rotkehlchen.chains_aggregator.zksync_lite.get_db_transactions()
-            else:
-                transactions = []
-                for tx_hash in entry['tx_hashes']:
-                    transactions.extend(  # yeah I know not performant but this getsalways called with 1 tx_hash  # noqa: E501
-                        self.rotkehlchen.chains_aggregator.zksync_lite.get_db_transactions(
-                            queryfilter=' WHERE tx_hash=?',
-                            bindings=(tx_hash,),
-                        ),
-                    )
 
-            for transaction in transactions:
-                self.rotkehlchen.chains_aggregator.zksync_lite.decode_transaction(
-                    transaction=transaction,
-                    tracked_addresses=tracked_addresses,
+        # first delete tranasaction data and all decoded events and related data
+        with self.rotkehlchen.data.db.user_write() as write_cursor:
+            concerning_address = write_cursor.execute('DELETE FROM zksynclite_transactions WHERE tx_hash=? RETURNING from_address', (tx_hash,)).fetchone()  # noqa: E501
+            deleted_event_data = write_cursor.execute(
+                'DELETE FROM history_events WHERE event_identifier=? RETURNING location_label',
+                (f'zkl{tx_hash.hex()}',),
```

Comment:
shouldn't the ZKL be a constant? or the `zkl{tx_hash}` a constant that you can later use to do `ZKL_IDENTIFIER.format(tx_hash=tx_hash.hex())`